packages/runtime/container-runtime/src/channelCollection.ts

```diff
@@ -621,11 +621,11 @@ export class ChannelCollection implements IFluidDataStoreChannel, IDisposable {
 	 */
 	protected createDataStoreId(): string {
 		/**
-		 * There is currently a bug where certain data store ids such as "[" are getting converted to ASCII characters
-		 * in the snapshot.
-		 * So, return short ids only if explicitly enabled via feature flags. Else, return uuid();
+		 * Return uuid if short-ids are explicitly disabled via feature flags.
 		 */
-		if (this.mc.config.getBoolean("Fluid.Runtime.IsShortIdEnabled") === true) {
+		if (this.mc.config.getBoolean("Fluid.Runtime.IsShortIdDisabled") === true) {
+			return uuid();
```

Comment:
I don't think this can be enabled by default due to cross-client compat concerns - Runtimes older than this that receive data store / DDS with short ids will end up corrupting the summary right?
Especially customers using public APIs will have versions older than 2.31.0 (where this is added) for a long time.

Reply:
With the PR: https://github.com/microsoft/FluidFramework/pull/24097, we are ensuring that we only use encode safe characters. So even if this latest runtime client generated a short id and if an older runtime-client (<2.30.0) encoded it, will not change it's value with `encodeURIComponent()`.

Reply:
oh yeah, I forgot that part.

Reply:
can you please add that to the description so other people don't get confused as well :-).